src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexPersistence.cs

```diff
@@ -198,6 +198,10 @@ public void Clean(CleanupMode mode)
             _converter?.Clean();
             _indexSearcherHolder.Cleanup(_index._indexStorage.Environment().PossibleOldestReadTransaction(null), mode);
 
+            var indexWriter = _indexWriter;
+            if (indexWriter != null)
+                indexWriter.RecreateAfterCommit = true;
```

Comment:
Why it this being set here only? When is this running? Shouldn't this be on deep mode, then?

Reply:
In general:
```
low mem -> regular clean up
timeToWaitForMemoryCleanup -> deep if no query for 10 min otherwise regular;
idleness -> can be deep if no work was for 5 min otherwise regular
```

Since this is a index _writer_ we clean it every time we reduce the mem. (right now we were clean it upon each commit)

Reply:
If no query happened in the last ten minutes, we lose the writer?
Same for idle for 5 minutes? 

That means that a write every 15 minutes would be very expensive, no?

Why are we cleaning this up if there is no pressure?

Reply:
We clean the writer if 
1. under low mem
2. idle run every 10 min (which is wired by its own, it seem we always clean every 10 min regardless)
3. no new docs for 5 seconds.

I think we don't need to over allocate here and clean up in reasonable time-frame and not wait for low mem to happened.
(mem pressure isn't the only reason for cleaning up, high number of managed objects is something we don't want either)

Reply:
The problem is that we will have high CPU after idle time, no? 
And 5 sec with no docs happens a _lot_.
Imagine a document coming in every 6 seconds. Then you are back to high CPU .

Reply:
I agree it might be a problem, the question is, when do we want to cleanup the writer.
should we return the previous behavior of ~250K writes?

Reply:
I would prefer to do that on low memory only.
250K writes isn't ideal, I think. We throw a lot of memory away then, which we then reuse?
Maybe after we run a merge, instead, since we'll need to reopen the new segment anyway?

Reply:
If I understand correctly, upon merge we always drop, so we shouldn't worry about that.

My concern is that we are not playing nice with the memory and will be forced to hit low-mem some day, instead of doing that per index once in a while and avoid the high costs of the low-mem event.

Reply:
The "once in a while" thing worries me.
I guess we could set it to be something like: "after 10 minutes of no writes (needs to be configurable)", dispose the writer.